File: April_2022/27_04_2022/safe.cpp
#include <iostream>
#include <math.h>

using namespace std;

int main(){
    
    int n,a,b;
    float x,y;
    
    cin>>n>>x>>y;
    
    for(int i=1;i<n;i++){
        float l = sqrt(pow(n,2)-pow(i,2));
        if(l == (int)l){
            a = max(i,(int)l);
            b = min(i,(int)l);
        }
    }
    
    if(abs(a-x)<=1 || abs(b-y)<=1){
        cout<<"Unsafe";
    }
    else{
        cout<<"Safe";
    }
    
    return 0;
}

File: April_2022/27_04_2022/safe.py
import math

n = int(input())
x,y = list(map(float,input().split()))
a,b = 0,0
for i in range(1,n):
    l = math.sqrt(n*2-i*2)
    if(l == int(l)):
        a = max(i,int(l))
        b = min(i,int(l))
        break
if(abs(x-a)<=1 or abs(y-b)<=1):
    print("Unsafe")
else:
    print("Safe")